File: Java OOP/Homework07/presenters/PersonPresenter.py
from models.Designer import Designer
from models.Person import Person
from models.Programmer import Programmer
from views import View

from presenters.DesignerPresenter import DesignerPresenter
from presenters.ProgrammerPresenter import ProgrammerPresenter


class PersonPresenter:
    def __init__(self, person: Person, view: View):
        self.person = person
        self.view = view
    
    def get_menu_item(self) -> int:
        print("1). Информация о сотруднике")
        print("0). Назад")

        return int(self.view.get_input())

    def menu_control(self):
        menu_index = -1

        while menu_index != 0:
            menu_index = self.get_menu_item()

            if menu_index == 1:
                self.print_person_info(None)
            elif menu_index == 0:
                pass
            else:
                self.view.print_message("Некорректный ввод")

    def print_person_info(self, programmer: Programmer):
        if isinstance(self.person, Programmer):
            programmer_view = ProgrammerPresenter(self.person, self.view)
            programmer_view.print_programmer_info()
        elif isinstance(self.person, Designer):
            designer_presenter = DesignerPresenter(self.person, self.view)
            designer_presenter.print_designer_info()

File: Java OOP/Homework06/presenters/PersonPresenter.java
package presenters;

import views.View;
import models.Designer;
import models.Person;
import models.Programmer;

public class PersonPresenter {

    Person person;
    View view;

    public PersonPresenter(Person person, View view) {
        this.person = person;
        this.view = view;
    }

    public int getMenuItem() {
        System.out.println("1). Информация о сотруднике");
        System.out.println("0). Назад");

        return Integer.parseInt(view.getInput());
    }

    public void menuControl() {
        int menuIndex;

        do {
            menuIndex = getMenuItem();

            switch (menuIndex) {
                case 1: // Информация о работнике
                    printPersonInfo(null);
                    break;

                case 0: // Удалить отдел
                    break;

                default:
                    view.printMessage("Некорректный ввод");
                    break;
            }
            menuIndex = getMenuItem();
        } while (menuIndex != 0);
    }

    public void printPersonInfo(Programmer programmer) {
        if (person instanceof Programmer) {
            ProgrammerPresenter programmerView = new ProgrammerPresenter((Programmer) person, view);
            ;
            programmerView.printProgrammerInfo();
        } else if (person instanceof Designer) {
            DesignerPresenter designerPresenter = new DesignerPresenter((Designer) person, view);
            ;
            designerPresenter.printDesignerInfo();
        }
    }

}